module/VuFind/src/VuFind/ILS/Driver/Aleph.php

```diff
@@ -1191,23 +1191,18 @@ public function cancelHolds($details)
         $count = 0;
         $statuses = [];
         foreach ($details['details'] as $id) {
-            $result = $this->doRestDLFRequest(
-                [
-                    'patron', $patronId, 'circulationActions', 'requests', 'holds',
-                    $id
-                ], null, "DELETE"
-            );
-            $reply_code = $result->{'reply-code'};
-            if ($reply_code != "0000") {
-                $message = $result->{'del-pat-hold'}->{'note'};
-                if ($message == null) {
-                    $message = $result->{'reply-text'};
-                }
-                $statuses[$id] = [
-                    'success' => false, 'status' => 'cancel_hold_failed',
-                    'sysMessage' => (string)$message
-                ];
-            } else {
+            try {
+                $result = $this->doRestDLFRequest(
+                    [
+                        'patron', $patronId, 'circulationActions', 'requests', 'holds',
+                        $id
+                    ], null, "DELETE"
+                );
+            } catch (AlephRestfulException $e) {
+                $statuses[$id]
+                    = ['success' => false, 'status' => 'cancel_hold_failed'];
```

Comment:
In the original version of the code, we returned the system message in the 'sysMessage' element of the return array, but that seems to have been dropped in the refactoring here. I believe you could bring it back with `'sysMessage' => $e->getMessage()` if you wanted it. Was it removed intentionally?

Reply:
Yes, you are right. At first, it was intentional. Now I see it still works with retrieving the message directly from the exception.